client/src/ContainsTemplate.cpp

```diff
@@ -9,20 +9,88 @@ using namespace std;
 using namespace cv;
 
 #include <iostream>
+#include <algorithm>
+
+// Find the actual image coords ignoring any black borders
+Rect FindImageRect(Mat &mask) {
+  assert(mask.type() == CV_8UC1);
+
+  Point a, b;
+
+  for (size_t i = 0; i < mask.rows; i++) {
```

Comment:
I recommend using `++i` in general, when writing C++. In this case it doesn't matter since `i` is a `size_t`, but more generally `i` will be an iterator and `++i` might be better 💃 💃